describe query parameter passing for public keys (#37)
w3c-ccg/did-pkh: did-pkh-method-draft.md

### Create

The blockchain account id is validated according to [CAIP-10][] and then
appended to `did:pkh:{network}:`, where `{network}` is the supported prefix
corresponding to the blockchain ecosystem within which it is valid. No DID
Document should be returned for a DID URL that is not a valid [CAIP-10][] URI.

> In some contexts, such as in web-based interfaces for Solana or Tezos wallets, a

Reviewer: ```suggestion
In some contexts, such as web-based interfaces for Solana or Tezos wallets, a
```

public key is traditionally passed by wallets to counterparties as part of
authentication, without which signatures cannot be verified and submitted to

> public ledgers. In other contexts, such as in web-based interfaces for Ethereum

Reviewer: ```suggestion
public ledgers. In other contexts, such as web-based interfaces for Ethereum
```

wallets, a public key is cryptographically [recoverable][EC Recovery for
secp256k1] from a signature over a known message, but unknowable without one. In
all of these cases, a DID URL containing only the public-key hash will not be

> enough to recover the corresponding public key and thus generate a useful DID

Reviewer: ```suggestion
enough to recover the corresponding public key and use it to generate a useful DID
```

> document; instead, a query parameter `pk` (i.e. `?pk=`) can be appended to a DID
> PKH URL such that a DID document can be generated for that DID PKH URL. For a
> valid DID URL (i.e. one conforming to [CAIP-10]) without the additional

Reviewer: ```suggestion
document; instead, the query parameter `pk` can be appended to a DID
PKH URL (i.e., `?pk=`), and a DID document can be generated for that DID PKH URL. For a
valid DID URL (i.e., one conforming to [CAIP-10]) without the additional
```

> information needed to generate a DID Document, an "empty" DID document

Reviewer: ```suggestion
information needed to generate a DID document, an "empty" DID document
```

## Appendix: Legacy Support

An earlier version of this specification used more human-readable submethod
namespacing rather than referring directly to the CAIP naming convention. It

> also defaulted to main-net for `did:pkh:eth` in the absence of an explicit

Reviewer: ```suggestion
also defaulted to `main-net` for `did:pkh:eth` in the absence of an explicit
```

> chainID.  As the scope has grown of this project, and with forward compatibility

Reviewer: ```suggestion
`chainID`.  As the scope of this project has grown, and with forward compatibility
```

> in mind, both of these patterns have been removed, required explicitly naming

Reviewer: ```suggestion
in mind, both of these patterns have been removed, now requiring explicitly naming
```

the EVM by its registered [CAIP-2][] code and explicitly naming the `chain_id`
(as specified in [CAIP-10][]) as well.  The legacy aliases that Spruce's

> implementation also supports for backwards-compatibility with credentials
> already issued look like this:

Reviewer: ```suggestion
implementation also supports for backwards-compatibility with previously
issued credentials look like this:
```
